dm/dm/master/scheduler/scheduler_test.go

```diff
@@ -481,6 +482,40 @@ func (t *testScheduler) testSchedulerProgress(c *C, restart int) {
 	t.relayStageMatch(c, s, sourceID2, pb.Stage_InvalidStage)
 	rebuildScheduler(ctx)
 
+	// CASE 4.7.1: add source2 with specify worker1
+	// source2 not exist, worker2 should be free
+	t.sourceCfgNotExist(c, s, sourceID2)
+	t.workerFree(c, s, workerName2)
+	c.Assert(s.AddSourceCfgWithWorker(&sourceCfg2, workerName1), IsNil)
+	// source2 is unbound because expected worker1 is already bound
```

Comment:
I'm not sure this behaviour is expected. In other words, if user want to add a source to a specific DM-worker, and the worker is not available (offline or already bound), should we bind the source to another worker, left the source unbound or remove the source?
Maybe you need this feature in your scenario, could you explain why you choose this bevaiour?

BTW, when DM-master failover, DM-master will try to bind and unbound sources.

Reply:
specific DM-worker only allow single source, user can do a transfer or remove the source quickly if the specific worker is not available, if we do something like bind the source to another worker (or remove it) can cause some extra "scheduling cost". And i think a "specify-worker" cmd bind a not specific worker may not be expected.

Reply:
> a "specify-worker" cmd bind a not specific worker may not be expected

I got this.

How about we don't add the source when the specified worker is not available? I can list two reason:
1. The command is more atomic, the effects all happen or nothing happen
2. When the DM-master failovers, currently we will try to bind any unbound source, so the source may be bound to a unwanted worker.

Reply:
ok,  i'll modify this part and ci later.